src/overmap.cpp

```diff
@@ -1020,22 +928,48 @@ bool overmap_special::can_be_placed_on( const oter_id &oter ) const
 
 bool overmap_special::requires_city() const
 {
-    return min_city_size > 0 || max_city_distance < std::max( OMAPX, OMAPY );
+    return city_size.min > 0 || city_distance.max < std::max( OMAPX, OMAPY );
 }
 
 bool overmap_special::can_belong_to_city( const tripoint &p, const city &cit ) const
 {
     if( !requires_city() ) {
         return true;
     }
-    if( !cit || cit.s < min_city_size || cit.s > max_city_size ) {
+    if( !cit || !city_size.contains( cit.s ) ) {
         return false;
     }
-    const int dist = cit.get_distance_from( p );
-    return dist >= min_city_distance && dist <= max_city_distance;
+    return city_distance.contains( cit.get_distance_from( p ) );
 }
 
-void overmap_special::check()
+void overmap_special::load( JsonObject &jo, const std::string & )
+{
+    mandatory( jo, was_loaded, "overmaps", terrains );
+
+    std::vector<std::string> tmp_locations;
+    mandatory( jo, was_loaded, "locations", tmp_locations );
+    for( const std::string &elem : tmp_locations ) {
+        const auto iter = special_locations.find( elem );
+
+        if( iter != special_locations.end() ) {
+            locations.insert( &iter->second );
+        } else {
+            debugmsg( "Overmap special \"%s\" has invalid location \"%s\".",
+                      id.c_str(), elem.c_str() );
+        }
+    }
+
+    mandatory( jo, was_loaded, "occurrences", occurrences );
+
+    optional( jo, was_loaded, "connections", connections );
+    optional( jo, was_loaded, "city_sizes", city_size );
+    optional( jo, was_loaded, "city_distance", city_distance );
+    optional( jo, was_loaded, "rotate", rotatable, false );
+    optional( jo, was_loaded, "spawns", spawns );
+    optional( jo, was_loaded, "flags", flags );
```

Comment:
`assign` appears to be incapable of working with complex types yet (other than `std::set<std::string>`). That leaves only two possible candidates for the function: `rotatable` and `flags`. Although, it can be updated to support `numeric_interval`s.